fahrschein/src/main/java/org/zalando/fahrschein/domain/Authorization.java

```diff
@@ -0,0 +1,63 @@
+package org.zalando.fahrschein.domain;
+
+import javax.annotation.concurrent.Immutable;
+import java.util.List;
+import java.util.Objects;
+
+@Immutable
+public class Authorization {
+
+    @Immutable
+    public static class AuthorizationAttribute {
+
+        public static final AuthorizationAttribute ANYONE = new AuthorizationAttribute("*", "*");
+
+        private final String dataType;
+        private final String value;
+
+        public AuthorizationAttribute(String dataType, String value) {
+            this.dataType = dataType;
+            this.value = value;
+        }
+
+        public String getDataType() {
+            return dataType;
+        }
+
+        public String getValue() {
+            return value;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o)
+                return true;
+            if (!(o instanceof AuthorizationAttribute))
+                return false;
+            AuthorizationAttribute that = (AuthorizationAttribute) o;
+            return dataType.equals(that.dataType) &&
+                    value.equals(that.value);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(dataType, value);
+        }
+    }
+
+    private final List<AuthorizationAttribute> admins;
+    private final List<AuthorizationAttribute> readers;
+
+    public Authorization(List<AuthorizationAttribute> admins, List<AuthorizationAttribute> readers) {
+        this.admins = admins;
```

Comment:
Should make defensive copies if you want to be immutable.

Reply:
Or use persistent, immutable collections https://github.com/GlenKPeterson/Paguro without exposing them.

Reply:
Yeah, I changed to `java.util.Collections.unmodifiableList`, because this is how it's done in other domain objects in this package already.